client/blocks/importer/wordpress/upgrade-plan/upgrade-plan-details.tsx

```diff
@@ -1,24 +1,220 @@
 import { recordTracksEvent } from '@automattic/calypso-analytics';
+import config from '@automattic/calypso-config';
 import {
+	calculateMonthlyPriceForPlan,
 	getPlan,
+	Plan,
 	PLAN_BUSINESS,
 	PLAN_BUSINESS_MONTHLY,
 	isMonthly,
 } from '@automattic/calypso-products';
 import { CloudLogo, Button, PlanPrice } from '@automattic/components';
+import { SitePlanPricing } from '@automattic/data-stores';
+import { useSitePlans } from '@automattic/data-stores/src/plans';
 import { Title } from '@automattic/onboarding';
 import { Plans2023Tooltip, useManageTooltipToggle } from '@automattic/plans-grid-next';
+import styled from '@emotion/styled';
 import { useI18n } from '@wordpress/react-i18n';
 import clsx from 'clsx';
+import { useTranslate } from 'i18n-calypso';
 import React, { useState, useEffect } from 'react';
 import ButtonGroup from 'calypso/components/button-group';
 import QuerySitePlans from 'calypso/components/data/query-site-plans';
 import { useSelectedPlanUpgradeMutation } from 'calypso/data/import-flow/use-selected-plan-upgrade';
+import { SitePlanData } from 'calypso/my-sites/checkout/src/hooks/product-variants';
 import { useSelector } from 'calypso/state';
-import { getSitePlan, getSitePlanRawPrice } from 'calypso/state/sites/plans/selectors';
+import { getSitePlan } from 'calypso/state/sites/plans/selectors';
 import { UpgradePlanFeatureList } from './upgrade-plan-feature-list';
 import { UpgradePlanHostingDetails } from './upgrade-plan-hosting-details';
 
+interface SwitcherProps {
+	selectedPlan: string;
+	onMonthlyPlanClick: () => void;
+	onAnnualPlanClick: () => void;
+}
+
+const UpgradePlanPeriodSwitcher = ( props: SwitcherProps ) => {
+	const { __ } = useI18n();
+
+	if ( config.isEnabled( 'migration-flow/introductory-offer' ) ) {
+		return null;
+	}
```

Comment:
Hmmm... I thought we would allow users to purchase monthly plans, but would not get the introductory offer in that case, and we'd just show the standard offer. But I could easily be wrong! This just surprised me.

Reply:
Yes, there were a few back and forth. 
Here is the final decision about it: p9Jlb4-c3O-p2#comment-11827

Reply:
Thanks for confirming! It will be really interesting to see if/how conversion gets impacted by removing the cheaper upfront option.